From 6421b14ecbde1361089d86998a652eb6ee2035a1 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 6 Apr 2016 10:24:58 +0200
Subject: [PATCH] plymouth/plymouth-pretrigger.sh: also trigger acpi subsystem

Trigger the acpi subsystem. This will ensure hv_vmbus gets loaded before
plymouth is started, which will make the graphics device become
available before plymouth is started too (and the keyboard ! which might
also be important for plymouth in some setups).

https://bugzilla.redhat.com/show_bug.cgi?id=1218130
(cherry picked from commit d2846fdcce9b8de0edecdf0e06a4b86fc8de542c)
---
 modules.d/50plymouth/plymouth-pretrigger.sh | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/modules.d/50plymouth/plymouth-pretrigger.sh b/modules.d/50plymouth/plymouth-pretrigger.sh
index c3a97f20..6458d78d 100755
--- a/modules.d/50plymouth/plymouth-pretrigger.sh
+++ b/modules.d/50plymouth/plymouth-pretrigger.sh
@@ -5,9 +5,14 @@ if type plymouthd >/dev/null 2>&1 && [ -z "$DRACUT_SYSTEMD" ]; then
         # first trigger graphics subsystem
         udevadm trigger --action=add --attr-match=class=0x030000 >/dev/null 2>&1
         # first trigger graphics and tty subsystem
-        udevadm trigger --action=add --subsystem-match=graphics --subsystem-match=drm --subsystem-match=tty >/dev/null 2>&1
+        udevadm trigger --action=add \
+            --subsystem-match=graphics \
+            --subsystem-match=drm \
+            --subsystem-match=tty \
+            --subsystem-match=acpi \
+            >/dev/null 2>&1
 
-        udevadm settle --timeout=30 2>&1 | vinfo
+        udevadm settle --timeout=180 2>&1 | vinfo
 
         info "Starting plymouth daemon"
         mkdir -m 0755 /run/plymouth
